tactics/equality.ml

```diff
@@ -429,18 +427,23 @@ let lookup_eq_eliminator_tc env sigma eq ~dep ~inccl ~l2r ~c_sort ~e_sort ~p_sor
   (sigma , c), indarg
 
 let lookup_eq_eliminator_with_error env sigma eq ~dep ~inccl ~l2r ~c_sort ~e_sort ~p_sort =
-  let is_global_exists gr c = match Rocqlib.lib_ref_opt gr with
-    | Some gr -> isRefX env sigma gr c
-    | None -> false in
-  let is_eq = is_global_exists "core.eq.type" eq in
-  let eq_scheme = eq_scheme_name dep l2r inccl (ESorts.quality sigma p_sort) (ESorts.is_set sigma p_sort) in
-  if is_eq && not (Option.is_empty eq_scheme) then
+  let is_equality_exists c = match Rocqlib.lib_ref_opt "core.eq.type" with
+    | Some gr -> if isRefX env sigma gr c then Option.make "eq"
+      else begin match Rocqlib.lib_ref_opt "core.identity.type" with
+        | Some gr -> if isRefX env sigma gr c then Option.make "identity" else None
+        | None -> None end
+    | None -> begin match Rocqlib.lib_ref_opt "core.identity.type" with
+        | Some gr -> if isRefX env sigma gr c then begin
+          Option.make "identity" end else None
+        | None -> None end in
+  let is_eq = is_equality_exists eq in
+  let eq_scheme () = eq_scheme_name (Option.get is_eq) dep l2r inccl (ESorts.quality sigma p_sort) (ESorts.is_set sigma p_sort) in
+  if not (Option.is_empty is_eq) && not (Option.is_empty @@ eq_scheme ()) then
```

Comment:
The mix of Option.is_empty and Option.get is terrible, please use pattern-matching properly.